token-swap/program/src/curve/calculator.rs

```diff
@@ -176,9 +79,6 @@ pub trait CurveCalculator: Debug + DynPack {
             .and_then(map_zero_to_none)
     }
 
-    /// Calculate the host fee based on the owner fee, only used in production
-    /// situations where a program is hosted by multiple frontends
-    fn host_fee(&self, _owner_fee: u128) -> Option<u128> {
-        Some(0)
-    }
+    /// Validate that the given curve has no bad parameters
+    fn validate(&self) -> Result<(), SwapError>;
```

Comment:
Currently used for bad offsets or constant prices.  We'll have to look into stable swap for bad amp factors, but that can be for another PR